### PR TITLE
@eessex => [Analytics] Add tracking to artwork grid on search, collect, collection and artist

### PR DESCRIPTION
**Pull request by mzikherman**
Tested locally, works great! The cohesion helpers were very smooth to use 😺

### CODE REVIEW

File: src/v2/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx [hunk 1]
```diff
@@ -40,6 +44,15 @@ const ArtistArtworkFilter: React.FC<ArtistArtworkFilterProps> = props => {
       aggregations={sidebarAggregations.aggregations as any}
       counts={artist.counts}
       onChange={updateUrl}
+      onArtworkBrickClick={artwork => {
+        tracking.trackEvent(
+          clickedMainArtworkGrid({
+            contextPageOwnerType: OwnerType.artist,
```
Comment (mzikherman) [inline src/v2/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx:50]: Should this be using `PageOwnerType`?

File: src/v2/Apps/Collect/Routes/Collect/index.tsx [hunk 2]
```diff
@@ -95,7 +98,16 @@ export const CollectApp = track({
               { value: "-year", text: "Artwork year (desc.)" },
               { value: "year", text: "Artwork year (asc.)" },
             ]}
-            onChange={(filters) => {
+            onArtworkBrickClick={artwork => {
+              tracking.trackEvent(
+                clickedMainArtworkGrid({
+                  contextPageOwnerType: OwnerType.collection, // TODO: Switch to 'collect'
```
Comment (mzikherman) [inline src/v2/Apps/Collect/Routes/Collect/index.tsx:104]: Should be 'collect' once Cohesion is updated.

File: src/v2/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx (hunk 1 above)
Comment (eessex) [reply src/v2/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx:50]: This is correct -- we only use `PageOwnerType` on the schema end.

File: src/v2/Apps/Collect/Routes/Collection/index.tsx [hunk 3]
```diff
@@ -139,6 +141,15 @@ export class CollectionApp extends Component<CollectionAppProps> {
                     artworksConnection.aggregations as SharedArtworkFilterContextProps["aggregations"]
                   }
                   onChange={updateUrl}
+                  onArtworkBrickClick={artwork => {
+                    tracking.trackEvent(
+                      clickedMainArtworkGrid({
+                        contextPageOwnerType: OwnerType.collection,
```
Comment (eessex) [inline src/v2/Apps/Collect/Routes/Collection/index.tsx:147]: We should include the `contextPageOwnerSlug` and `contextPageOwnerId` here as well.

File: src/v2/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx (hunk 1 above)
Comment (eessex) [reply src/v2/Apps/Artist/Routes/Overview/Components/ArtistArtworkFilter.tsx:50]: We should include optional properties `contextPageOwnerSlug` and `contextPageOwnerId` here.

File: src/v2/Apps/Collect/Routes/Collection/index.tsx (hunk 3 above)
Comment (mzikherman) [reply src/v2/Apps/Collect/Routes/Collection/index.tsx:147]: Yup, good call.

**Comment by mzikherman** [comment]
Updated with correct tracking for collect, and adding ids and slugs to the artist and collection page tracking.

**Pull request merged**